File: app/src/androidTest/java/com/example/finalproject/AddData.java
package com.example.finalproject;

import android.util.Log;
import com.amplifyframework.core.Amplify;
import com.amplifyframework.datastore.generated.model.Post;

public class AddData {
    Post post = Post.builder()
            .title("Create an Amplify DataStore app")
            .status(PostStatus.ACTIVE)
            .build();

    Amplify.DataStore.save(post,
        result -> Log.i("MyAmplifyApp", "Created a new post successfully"),
        error -> Log.e("MyAmplifyApp",  "Error creating post", error)
                );

File: data_pipeline/add_data.dart
import 'package:amplify_flutter/amplify.dart';
import 'package:finalproject/models/Post.dart'; // Import the correct model class
import 'package:finalproject/models/PostStatus.dart'; // Import the correct status class
import 'dart:developer' as developer;

class AddData {
  void addPost() {
    final post = Post(
      title: "Create a Flutter DataStore app",
      status: PostStatus.ACTIVE,
    );

    Amplify.DataStore.save(
      post,
      onSuccess: (savedItem, {mutatedItems}) =>
          developer.log('Created a new post successfully', name: 'MyAmplifyApp'),
      onError: (error) =>
          developer.log('Error creating post: $error', name: 'MyAmplifyApp'),
    );
  }
}
